app/models/concerns/host_aspects/managed_host_extensions.rb

```diff
@@ -0,0 +1,75 @@
+require 'host_aspects'
+
+module HostAspects
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      HostAspects::ManagedHostExtensions.refresh_aspect_relations(self)
+
+      has_many :host_aspects, :foreign_key => :host_id, :inverse_of => :host do
+        def [](subject)
+          where(:aspect_subject => subject).first ||
+          select { |aspect| aspect.aspect_subject == subject}.first ||
+          build(:aspect_subject => subject)
+        end
+      end
+      accepts_nested_attributes_for :host_aspects
+
+      def attributes
```

Comment:
Have you tried saving an object with attributes extended in such a way? ActiveRecord may not be happy...